scripts/area_plot.py

```diff
@@ -0,0 +1,613 @@
+import argparse
+import warnings
+# from warnings import Warning
+
+from matplotlib import rcParams
+from matplotlib import cm
+import matplotlib.pyplot as plt
+import numpy as np
+import pandas as pd
+
+"""
+A set of python scripts to generate an area plot.
+"""
+# Sets up the matplotlib paramters so that we can save to be edited in 
+# illustator if a direct conversion is required. Because it just makes life
+# better
+rcParams['pdf.fonttype'] = 42
+rcParams['ps.fonttype'] = 42
+
+# Sets up the order of colors to be used in joint plots.
+colors_order = ['Reds', 'Blues', 'Greens', 'Purples', "Oranges", 'Greys']
+
+over9 = {'Paired', 'Paired_r', 'Set3', 'Set3_r'}
+over8 = over9 | {'Set1', "Pastel1"}
+
+def extract_label_array(table, tax_col, tax_delim='|'):
+    """
+    Converts delimited taxonomy strings into a working table
+    
+    Parameters
+    ----------
+    table : DataFrame
+        A DataFrame with observation on the rows (biom-style table) with 
+        `tax_col` as one of its columns.
+    tax_col : str
+        The column in `table` containig the taxobnomy information
+    tax_delim: str, optional
+        The delimiter between taxonomic groups
+        
+    Returns
+    -------
+    DataFrame
+        The taxonomic strings parsed into n levels
+    """
+    def f_(x):
+        return pd.Series([y.strip() for y in x.split(tax_delim)])
+    
+    return table[tax_col].apply(f_)
+
+
+def level_taxonomy(table, taxa, samples, level, consider_nan=True):
+    """
+    Gets the taxonomy collapsed to the desired level
+    
+    Parameters
+    ----------
+    table : DataFrame
+        A table with observation on the rows (biom-style table) with `samples`
+        in its columns.
+    taxa: DataFrame
+        The taxonomic strings parsed into n levels
+    level: list
+        The level to which the taxonomy should be summarized 
+    samples : list
+        The columns from `table` to be included in the analysis
+    consider_nan: bool, optional
+        Whether the table contains multiple concatenated, in which cases 
+        considering `nan` will filter the samples to retain only the levels 
+        of interest. This is recommended for kracken/bracken tables, but not 
+        applicable for some 16s sequences
+    """
+    level = level.max()
+    print(level)
```

Comment:
Not really needed. If you want diagnostic output we should consider putting some `logging` in.

Reply:
Delightful diagnostic code. >_<